src/app/dashboard/user-administration/user-administration.component.ts

```diff
@@ -0,0 +1,140 @@
+} from 'rxjs';
+import { catchError, map, startWith, switchMap } from 'rxjs/operators';
+import { MatTableModule } from '@angular/material/table';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { DatePipe, TitleCasePipe } from '@angular/common';
+import { MatIcon } from '@angular/material/icon';
+import { MatButtonToggle } from '@angular/material/button-toggle';
+import { FormControl, FormsModule } from '@angular/forms';
+import {
+  MatFormField,
+  MatInput,
+  MatLabel,
+  MatPrefix,
+  MatSuffix,
+} from '@angular/material/input';
+import { MatButton, MatIconButton } from '@angular/material/button';
+import { RoundedInputComponent } from '@app/shared/rounded-input/rounded-input.component';
+import { UserAdministrationService } from '@app/dashboard/user-administration/user-administration.service';
+import { User } from '@app/auth/auth.type';
+import { MatChip } from '@angular/material/chips';
+import { MatMenu, MatMenuItem, MatMenuTrigger } from '@angular/material/menu';
+import { MatDialog } from '@angular/material/dialog';
+import { AddUserComponent } from '@app/dashboard/user-administration/add-user/add-user.component';
+
+@Component({
+  selector: 'app-user-administration',
+  styleUrl: 'user-administration.component.scss',
+  templateUrl: 'user-administration.component.html',
+  standalone: true,
+  imports: [
+    MatProgressSpinnerModule,
+    MatTableModule,
+    MatSortModule,
+    MatPaginatorModule,
+    DatePipe,
+    MatIcon,
+    MatButtonToggle,
+    FormsModule,
+    MatInput,
+    MatSuffix,
+    MatIconButton,
+    MatFormField,
+    MatLabel,
+    MatPrefix,
+    RoundedInputComponent,
+    MatButton,
+    MatChip,
+    TitleCasePipe,
+    MatMenuTrigger,
+    MatMenu,
+    MatMenuItem,
+  ],
+})
+export class UserAdministrationComponent implements AfterViewInit, OnDestroy {
+  displayedColumns: string[] = [
+    'created_at',
+    'name',
+    'email',
+    'role',
+    'status',
+    'more',
+  ];
+
+  data: User[] = [];
+
+  totalItems = 0;
+  isLoadingResults = true;
+  searchValue = new FormControl('');
+  private readonly destroy = new Subject<void>();
+  page = new FormControl(1);
+
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @ViewChild(MatSort) sort!: MatSort;
+
+  constructor(
+    private readonly userAdministrationService: UserAdministrationService,
+    private readonly dialog: MatDialog
+  ) {}
+
+  ngAfterViewInit() {
+    combineLatest([
+      this.searchValue.valueChanges.pipe(
+        startWith(''),
+        filter((searchValue): searchValue is string => searchValue !== null)
+      ),
+      this.page.valueChanges.pipe(startWith(1)),
+      this.paginator.page.pipe(startWith(new PageEvent())),
+    ])
+      .pipe(
+        takeUntil(this.destroy),
+        debounceTime(1000),
+        switchMap(([search, sort, page]) => {
+          console.log(sort, 'sort');
+          console.log(page, 'paginator');
+          console.log(search, 'search');
```

Comment:
Logs???

Reply:
Yeah still need them because of the filter functionality which is not ready on the UI yet. So for now the logs is fine, if not I will get eslint errors for not using them